libbeat/publisher/queue/spool/codec_test.go

```diff
@@ -0,0 +1,74 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package spool
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/publisher"
+)
+
+func TestEncodeDecode(t *testing.T) {
+	tests := map[string]codecID{
+		"jsob":   codecJSON,
+		"ubjson": codecUBJSON,
+		"cborl":  codecCBORL,
+	}
+
+	event := publisher.Event{
+		Content: beat.Event{
+			Timestamp: time.Now().Round(0),
+			Fields: common.MapStr{
+				"time":       time.Now().UTC().Round(time.Millisecond),
```

Comment:
(here and below) Minor in this case since the test doesn't really rely on it, but tests should be deterministic: use fixed `time.Time` values as initial inputs rather than `time.Now()`